### PR TITLE
field endpoints

### PR DESCRIPTION
**Pull request by jjidt**
- document/field endpoints
- field instance methods for value setting/getting
- monitored database to ensure values were getting cascade deletion on field deletion

### CODE REVIEW

File: api/src/controllers/documentFields.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    res.status(e.code || 500).json({ error: `error fetching document with id ${id}` })
+  }
+}
+
+const addDocumentField = async function(req, res) {
+  const { body } = req
+  const { projectId, documentId } = req.params
+  const { orgId } = req.requestorInfo
+  // Overwrite orgId even if they passed anything in
+  body.orgId = orgId
```
Comment (dheniges) [inline api/src/controllers/documentFields.js:89]: I recall this code somewhere else, eventually that should probably be an error message if they pass an invalid org id? This works for now though.

File: api/src/db/__testSetup__/initializeTestDb.js [hunk 2]
```diff
@@ -8,7 +8,9 @@ afterAll(() => {
 export const clearDb = async function() {
   const db = getDb()
 
-  await db.truncate()
+  await Promise.all(Object.values(db.models).map((model) => {
+    return model.destroy({ truncate: true, cascade: true })
```
Comment (dheniges) [inline api/src/db/__testSetup__/initializeTestDb.js:12]: What does this do that truncate() wasn't?

File: api/src/db/models/field.js [hunk 3]
```diff
@@ -1,16 +1,59 @@
...
+  }
+
+  Field.prototype.setFieldValue = function(fieldValue) {
+    const modelName = fieldTypes.FIELD_TYPE_MODELS[this.type]
+    return this[`set${modelName}`](fieldValue)
+  }
+
+  Field.prototype.getFieldValue = function(fieldValue) {
+    const modelName = fieldTypes.FIELD_TYPE_MODELS[this.type]
+    return this[`get${modelName}`](fieldValue)
```
Comment (dheniges) [inline api/src/db/models/field.js]: isn't `fieldValue` here and two lines above always going to be undefined since this is a getter?

**Review by dheniges — CHANGES_REQUESTED** [review]
JJ you've created a masterpiece. This is much more straightforward than I was anticipating, thanks to those pesky relationships seeming to do their job.

Looks like an unnecessary arg on a getter, otherwise it's solid

File: api/src/db/__testSetup__/initializeTestDb.js (hunk 2 above)
Comment (jjidt) [reply api/src/db/__testSetup__/initializeTestDb.js:12]: after defining the value/field relationship with the cascading delete, it started to throw an error message on truncate, this fixed it

File: api/src/db/models/field.js (hunk 3 above)
Comment (jjidt) [reply api/src/db/models/field.js]: oops, yeah copypasta

File: api/src/controllers/documentFields.js (hunk 1 above)
Comment (jjidt) [reply api/src/controllers/documentFields.js:89]: yeah, we're doing this elsewhere, we could probably tackle all those in a PR

**Review by dheniges — APPROVED** [review]

**Pull request merged**